app/assets/javascripts/admin/enterprises/controllers/enterprise_controller.js.coffee

```diff
@@ -21,6 +21,26 @@ angular.module("admin.enterprises")
     $scope.cancel = (destination) ->
       $window.location = destination
 
+    $scope.removeLogo = ->
+      Enterprises.removeLogo($scope.Enterprise).then (data) ->
+        $scope.Enterprise = angular.copy(data)
+        $scope.$emit('enterprise:updated', $scope.Enterprise)
+
+        StatusMessage.display('success', t('admin.enterprises.remove_logo.removed_logo_successfully'))
+      , (response) ->
+        if response.data.error?
+          StatusMessage.display('failure', response.data.error)
+
+    $scope.removePromoImage = ->
+      Enterprises.removePromoImage($scope.Enterprise).then (data) ->
+        $scope.Enterprise = angular.copy(data)
```

Comment:
I am curious, why do you new this copy?

Reply:
@luisramos0 I did this to be safe and make sure changes to `$scope.Enterprise` do not affect other callbacks, considering `$scope.Enterprise` is bound to the view and we are caching a pristine copy of the enterprise in `Enterprises`.

```ruby
var deferred = $q.defer();

deferred.promise.then(function(data) {
  $scope.first = data;
});

deferred.promise.then(function(data) {
  data.value = "updated value";
  $scope.second = data;
});

deferred.resolve({value: "original value"});

// $scope.first.value => "updated value"
// $scope.second.value => "updated value"
```